src/components/Homepage/HomepageInsightCard.tsx

```diff
@@ -0,0 +1,175 @@
+          {maybePluralize(actionItemInsight.ruleIds.length, 'action item')} due
+          by {moment(actionItemInsight.targetDate).local().format('MMM Do')}
+          {!isNil(entity) && (
+            <>
+              {' '}
+              for{' '}
+              <EntityRefLink entityRef={entityRef} className={linkClasses.root}>
+                {entity.name}
+              </EntityRefLink>
+            </>
+          )}{' '}
+          {!isNil(initiative) && (
+            <>
+              {' '}
+              for the{' '}
+              <Link
+                to={initiativeRef({
+                  id: initiative.id.toString(),
+                })}
+                className={linkClasses.root}
+              >
+                {initiative.name}
+              </Link>{' '}
+              Initiative
+            </>
+          )}
+        </InsightCard>
+      );
+    }
+    case UserInsightType.FAILING_RULE: {
+      const failingRuleInsight = insight as FailingRuleUserInsight;
+      const entity = entities?.find(
+        currEntity => currEntity.id === failingRuleInsight.entityId,
+      );
+      const scorecard = scorecards.find(
+        scorecard => scorecard.id === failingRuleInsight.scorecardId,
+      );
+
+      const contentSuffix = failingRuleInsight.nextLadderLevel
+        ? ` to reach ${failingRuleInsight.nextLadderLevel}`
+        : '';
+
+      const entityRef = {
+        kind: 'Component',
+        namespace: 'default',
+        name: entity?.name || '',
+      };
+      console.log(entity);
+      console.log(scorecard);
+      return (
+        <InsightCard>
+          {maybePluralize(failingRuleInsight.ruleIds.length, 'rule')} left
+          {!isNil(entity) && (
+            <>
+              {' '}
+              for{' '}
+              <EntityRefLink entityRef={entityRef} className={linkClasses.root}>
+                {entity.name}
+              </EntityRefLink>
+            </>
+          )}
+          {contentSuffix}
+          {!isNil(scorecard) && (
+            <>
+              {' '}
+              in{' '}
+              <ScorecardRefLink
+                scorecardId={scorecard.id}
+                className={linkClasses.root}
+              >
+                {scorecard.name}
+              </ScorecardRefLink>
+            </>
+          )}
+        </InsightCard>
+      );
+    }
+    case UserInsightType.NO_ENTITIES_OWNED: {
+      return (
+        <InsightCard>
+          Looks like you own no entities. Try creating one or add yourself as an
+          owner to an existing one.
+        </InsightCard>
+      );
+    }
+    case UserInsightType.NO_FAILING_RULES: {
+      return (
+        <InsightCard>
+          Great work! You don't have any more Scorecard rules left; all your
+          services are crushing it! 🎉
+        </InsightCard>
+      );
+    }
+    case UserInsightType.NO_SCORECARDS: {
+      return (
+        <InsightCard>
+          Looks like you have no scorecards - create one here!
```

Comment:
Should "here" link somewhere?

Reply:
good catch, changed to `Looks like you have no scorecards - create one!`